`TestKube/ExecInWeb` flakiness

## Failure

#### Link(s) to logs

- https://github.com/gravitational/teleport/actions/runs/13019155652/job/36315590982?pr=51135

#### Relevant snippet

```
    testing.go:1399: race detected during execution of test
    --- FAIL: TestKube/ExecInWeb (2.43s)
FAIL integration.TestKube/ExecInWeb (2.43s)
```

--- a/lib/kube/proxy/watcher.go
+++ b/lib/kube/proxy/watcher.go
@@ -21,6 +21,7 @@ package proxy
 import (
 	"context"
 	"maps"
+	"slices"
 	"sync"
 	"time"
 
@@ -238,10 +239,11 @@ func (s *TLSServer) unregisterKubeCluster(ctx context.Context, name string) erro
 
 	// close active sessions before returning.
 	s.fwd.mu.Lock()
-	sessions := maps.Values(s.fwd.sessions)
+	// collect all sessions to avoid holding the lock while closing them
+	sessions := slices.Collect(maps.Values(s.fwd.sessions))
 	s.fwd.mu.Unlock()
 	// close active sessions
-	for sess := range sessions {
+	for _, sess := range sessions {
 		if sess.ctx.kubeClusterName == name {
 			// TODO(tigrato): check if we should send errors to each client
 			errs = append(errs, sess.Close())

